src/server/req_quejob.c

```diff
@@ -1395,7 +1395,7 @@ req_jobscript(struct batch_request *preq)
 #else /* server - server - server - server */
 	/* add the script to the job */
 	size = get_bytes_from_attr(&attr_jobscript_max_size);
-	if (preq->rq_ind.rq_jobfile.rq_size > size){
+	if (pj->ji_qs.ji_un.ji_newt.ji_scriptsz + preq->rq_ind.rq_jobfile.rq_size > size){
```

Comment:
This will still reject in that case.

`pj->ji_qs.ji_un.ji_newt.ji_scriptsz` is the size of the script that has made it to the server so far (zero on the first chunk)
`preq->rq_ind.rq_jobfile.rq_size` is the size of the chunk that is being received now

If `jobscript_max_size` is less than `SCRIPT_CHUNK_Z` then you'll error out on the first chunk of the script, if it's greater than `SCRIPT_CHUNK_Z` you'll error out on some later chunk of the script.

Reply:
Thank you for explanation, looks good to me.